samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -46,25 +53,91 @@ public RemoteApplicationRunner(Config config) {
    * Run the {@link StreamApplication} on the remote cluster
    * @param app a StreamApplication
    */
-  @Override public void run(StreamApplication app) {
+  @Override
+  public void run(StreamApplication app) {
     try {
-      // 1. build stream graph
-      StreamGraph streamGraph = new StreamGraphImpl(this, config);
-      app.init(streamGraph, config);
+      // 1. initialize and plan
+      ProcessorGraph processorGraph = getExecutionPlan(app);
 
-      // 2. create the physical execution plan
-      ExecutionPlanner planner = new ExecutionPlanner(config);
-      ProcessorGraph processorGraph = planner.plan(streamGraph);
+      // 2. create the necessary streams
+      List<StreamSpec> streams = processorGraph.getIntermediateStreams().stream()
+          .map(streamEdge -> streamEdge.getStreamSpec())
+          .collect(Collectors.toList());
+      StreamManager streamManager = new StreamManager(new JavaSystemConfig(config).getSystemAdmins());
+      streamManager.createStreams(streams);
 
       // 3. submit jobs for remote execution
       processorGraph.getProcessorNodes().forEach(processor -> {
           Config processorConfig = processor.generateConfig();
-          log.info("Starting processor {} with config {}", processor.getId(), config);
+          log.info("Starting processor {} with config {}", processor.getId(), processorConfig);
           JobRunner runner = new JobRunner(processorConfig);
           runner.run(true);
         });
     } catch (Throwable t) {
-      throw new SamzaException("Fail to run application", t);
+      throw new SamzaException("Failed to run application", t);
     }
   }
+
+  @Override
+  public void kill(StreamApplication app) {
+    try {
+      ProcessorGraph processorGraph = getExecutionPlan(app);
+
+      processorGraph.getProcessorNodes().forEach(processor -> {
+          Config processorConfig = processor.generateConfig();
+          log.info("Killing processor {} with config {}", processor.getId(), processorConfig);
```

Comment:
Fair enough. Less is more. I'm going to omit the config for now. I think we use the console appender for the JobRunner/ApplicationRunner, so most of this is just for debugging anyway.